x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/support/SimpleKdcLdapServer.java

```diff
@@ -0,0 +1,232 @@
+        simpleKdc.getKdcConfig().setLong(KdcConfigKey.MAXIMUM_RENEWABLE_LIFETIME, 604800000L);
+        logger.info("MINIMUM_TICKET_LIFETIME changed from {}  to {}", minimumTicketLifeTime,
+                simpleKdc.getKdcConfig().getMinimumTicketLifetime());
+        logger.info("MAXIMUM_RENEWABLE_LIFETIME changed from {}  to {}", maxRenewableLifeTime,
+                simpleKdc.getKdcConfig().getMaximumRenewableLifetime());
+        simpleKdc.init();
+        simpleKdc.start();
+    }
+
+    public String getRealm() {
+        return realm;
+    }
+
+    public int getLdapListenPort() {
+        return ldapPort;
+    }
+
+    public int getKdcPort() {
+        return kdcPort;
+    }
+
+    /**
+     * Creates a principal in the KDC with the specified user and password.
+     *
+     * @param principal principal name, do not include the domain.
+     * @param password password.
+     * @throws Exception thrown if the principal could not be created.
+     */
+    public synchronized void createPrincipal(final String principal, final String password) throws Exception {
+        simpleKdc.createPrincipal(principal, password);
+    }
+
+    /**
+     * Creates multiple principals in the KDC and adds them to a keytab file.
+     *
+     * @param keytabFile keytab file to add the created principals. If keytab file
+     *            exists and then always appends to it.
+     * @param principals principals to add to the KDC, do not include the domain.
+     * @throws Exception thrown if the principals or the keytab file could not be
+     *             created.
+     */
+    @SuppressForbidden(reason = "Uses Apache Kdc which requires usage of java.io.File")
+    public synchronized void createPrincipal(final Path keytabFile, final String... principals) throws Exception {
+        simpleKdc.createPrincipals(principals);
+        for (String principal : principals) {
+            simpleKdc.getKadmin().exportKeytab(keytabFile.toFile(), principal);
+        }
+    }
+
+    /**
+     * Stop Simple Kdc Server
+     * 
+     * @throws PrivilegedActionException
+     */
+    public synchronized void stop() throws PrivilegedActionException {
+        AccessController.doPrivileged(new PrivilegedExceptionAction<Void>() {
+
+            @Override
+            @SuppressForbidden(reason = "set or clear system property krb5 debug in tests")
+            public Void run() throws Exception {
+                if (simpleKdc != null) {
+                    try {
+                        simpleKdc.stop();
+                    } catch (KrbException e) {
+                        throw ExceptionsHelper.convertToRuntime(e);
+                    } finally {
+                        System.setProperty("sun.security.krb5.debug", Boolean.toString(krb5DebugBackupConfigValue));
+                    }
+                }
+
+                try {
+                    // Will be fixed in next Kerby version.
```

Comment:
Yuck. Can you add a link to the bug and why we need to do this? Also, is there something we can poll to see if we really need to continue to wait? If so, we can use `assertBusy` or `awaitBusy` so we only wait as long as wee need to and don't hit failures on slow CI machines.